pages/api/models/component.ts

```diff
@@ -0,0 +1,32 @@
+import { model, Schema, Types } from "mongoose";
+
+enum ComponentType {
+    TEXT = "text",
+    VIDEO = "video",
+    AUDIO = "audio",
+}
+
+interface ComponentInterface {
+    _id: Types.ObjectId;
+    type: ComponentType;
+    properties: any;
```

Comment:
might just want to make an empty type for properties and so we don't forget to typecheck

Reply:
the front-end is going to pass in a JSON object for the component's styling and the back-end treat it as a black box - it just needs to retrieve the same JSON object when the front-end requests for it, front-end will take care of validating the JSON blob